private/bufpkg/bufconfig/write.go

```diff
@@ -273,30 +257,173 @@ func writeConfig(
 	for _, option := range options {
 		option(writeConfigOptions)
 	}
-	if !writeConfigOptions.documentationComments && writeConfigOptions.uncomment {
-		return errors.New("cannot set uncomment without documentationComments for WriteConfig")
+	if err := validateWriteConfigOptions(writeConfigOptions); err != nil {
+		return err
 	}
-	if writeConfigOptions.moduleIdentity == nil && len(writeConfigOptions.dependencyModuleReferences) > 0 {
-		return errors.New("cannot set deps without a name for WriteConfig")
+	// This is the same default as the bufconfig getters.
+	version := V1Beta1Version
```

Comment:
This seems wrong for the default value for write, are we sure about this?

Reply:
Yeah, this is consistent with the behaviour for when we go to get configs, if none are found, we default to `v1beta1` right now: https://github.com/bufbuild/buf/blob/f4fd40a47101848d722041ee6bfec433621bf1fe/private/bufpkg/bufconfig/get.go#L48-L50

There is a note to change this when we move to `v1`.